great_expectations_cloud/agent/models.py

```diff
@@ -4,17 +4,15 @@
 from typing import Any, Dict, Literal, Optional, Sequence, Set, Union
 from uuid import UUID
 
-from great_expectations.compatibility.pydantic import BaseModel, Extra, Field
+from great_expectations.compatibility.pydantic import BaseModel, ConfigDict, Field
 from great_expectations.experimental.metric_repository.metrics import MetricTypes
 from typing_extensions import Annotated
 
 from great_expectations_cloud.agent.exceptions import GXCoreError
 
 
 class AgentBaseModel(BaseModel):  # type: ignore[misc] # BaseSettings is has Any type
-    class Config:
-        # 2024-03-04: ZEL-501 Strictly enforce models for handling outdated APIs
-        extra: str = Extra.forbid
+    model_config = ConfigDict(extra="ignore")
```

Comment:
This seems dangerous to do this at the level of the `AgentBaseModel`.

Why not do it for the specific models you want this new behavior for?

Or create 2 different `AgentBaseModels`

```
AgentBaseIgnoreExtra
AgentBaseForbitExtra
```

Reply:
Łukasz had suggested that we do ignore the extra fields here: https://greatexpectationslabs.slack.com/archives/C05R8RE43CL/p1720051367833419?thread_ts=1720046306.191159&cid=C05R8RE43CL

This does make it harder for us to add more fields but it does ensure that we are not accepting in random inputs from messages.

Reply:
Also a helpful thread here: 
https://greatexpectationslabs.slack.com/archives/C04KWBU1XDF/p1718134632349739

Will merge this PR to unblock the work, but happy to talk more next week.

Reply:
If we want to ignore fields for certain models please create 2 separate base models and start inheriting from the `AgentBaseIgnoreExtra` where needed but otherwise inherit from `AgentBaseForbidExtra`.

Reply:
I'm against this.  It feel like now is not the right time to put in these classes. Could risk encroaching on over-engineering

Reply:
Renaming a model is not over-engineering.